src/sys/bout_types.cxx

```diff
@@ -52,6 +53,36 @@ const std::string& DIRECTION_STRING(DIRECTION direction) {
   return safeAt(DIRECTIONtoString, direction);
 }
 
+void swap(DirectionTypes& first, DirectionTypes& second) {
+  DirectionTypes temp = first;
+  first = second;
+  second = temp;
+}
+
+bool compatibleDirections(const DirectionTypes& d1, const DirectionTypes& d2) {
+  if (d1.y == d2.y && d1.z == d2.z) {
+    // direction types are the same, most common case, return immediately
+    return true;
+  }
+
+  if (d2.z == ZDirectionType::Average && d2.y == YDirectionType::Standard
+      && (d1.y == YDirectionType::Standard || d1.y == YDirectionType::Aligned)) {
```

Comment:
This second part will always be true. Can we just combine these two `if` statements?

Reply:
We could at the moment, but I think this way is stricter in case we ever add more `YDirectionType`s. I think actually I should check `d1.z == ZDirectionType::Standard` here as well, to be as strict as possible.